README.md

```diff
@@ -438,6 +438,34 @@ defines the following functions:
   elimination leading up to the results passed to this function.
 
 
+### `constexpr_wrapper`: function arguments as constant expressions
+
+The header
+```c++
+#include <vir/constexpr_wrapper.h>
+```
+defines the following tools:
+
+* `vir::constexpr_value` (concept): Satisfied by any type with a static 
+  `::value` member that can be used in a constant expression.
+
+* `vir::constexpr_wrapper<auto>` (class template): A type storing the value of 
+  its NTTP (non-type template parameter) and overloading all operators to 
+  return another `constexpr_wrapper`. `constexpr_wrapper` objects are 
+  implicitly convertible to their value type (a `constexpr_wrapper` 
+  automatically unwraps its constant expression).
+
+* `vir::cw<auto>` (variable template): Shorthand for producing 
+  `constexpr_wrapper` objects with the given value.
```

Comment:
This is really short. Do you want to have this directly in the `vir::` namespace?

Reply:
Yes. Really short is the *main* goal. I'd prefer to have it as short as writing nothing at all. But that requires a C++ language feature that has not progressed recently.

Reply:
I have a problem with "unreadable" public APIs in general. I like "being explicit is better". And that also means "being verbose" a bit to me.

If people do a `using cw = vir::constexpr_wrapper_v;` in their private .cxx, then I am totally fine with that.

Reply:
Yes, you're probably correct wrt. `vir::cw`. The goal here is to establish new common vocabulary something that pops up all over the place in C++. That's the goal for `std::cw` (or whatever it's going to be called. `std::c` gets people scared.... `std::const` clashes with the keyword). That won't happen with `vir::cw`. The question boils down to "should I use vir-simd as a testing ground for standard features or not"... :shrug: